### DISCUSSION TITLE
no audio on Acer CB3-532 "Banon"

### DISCUSSION THREAD

**Discussion by ksandbergfl** [sebanc/brunch#1443] [General]
I have Brunch running on an Acer CB3-532 "Banon" with Celeron N3060.  Everything seems to work great, except for audio.    This is a native Chromebook... so I am confused about the OS switches I should use, if any.   I have tried "native_chromebook_image" and "chromebook_audio".. as well as several different kernel versions.   Can't seem to get audio to work.... any tips/hints?

**Comment by tejasraman** [comment]
Try 4.4-chromebook kernel and chromebook_audio in ChromeOS(settings)
FYI, if you fix it, I have a similar chromebook(Celes) that needs brunch, tell me?

**Reply by ksandbergfl** [reply]
No, that doesn't work.  That was the first thing I tried.    I suspect that, because this is an actual Chromebook, there is another flag that I should be using too....

**Comment by tejasraman** [comment]
You know, I asked @sebanc about my Celes and he said try brunch unstable with 4.19 and that option(chromebook_audio)
[https://github.com/sebanc/brunch/discussions/1360](https://github.com/sebanc/brunch/discussions/1360)

**Comment by ksandbergfl** [comment]
I put Brunch in verbose boot mode, and I clearly see the problem -- when I select "chromebook_audio", and Brunch rebuilds the ChromeOS filesystems... there is an error "/rootc/packages/chromebook_audio/banon.tar.gz NOT FOUND"

**Comment by ksandbergfl** [comment]
I mounted the Brunch boot USB on another OS, and manually copied the "/rootc/packages/chromebook_audio/celes.tar.gz" to "/rootc/packages/chromebook_audio/banon.tar.gz" (the CELES and BANON are both Braswell-based) ---- and it WORKED!  I now have audio on my Brunchbook.

**Reply by sebanc** [reply]
This build should have direct support for banon board with "chromebook_audio" option:
https://github.com/sebanc/brunch-unstable/releases/tag/r98-unstable-20220219

**Comment by tejasraman** [comment]
Thanks! Yay! My CELES will work! Will see if it does, I will copy another Braswell Chromebook’s files if not.
Thanks again!

**Comment by tejasraman** [comment]
Ugh... My CELES has no audio under Brunch and I tried to use chromebook_audio (no success)

**Reply by ksandbergfl** [reply]
If you are using Brunch on a real Chromebook, you also have to select "native_Chromebook" or "native_chromebook_image" (I forget what the parameter is called.

**Reply by tejasraman** [reply]
No... the alsaucm configurations were messed up. While trying to set the UCM, I got an error message about "no Syntax line in chtrt5650" (Realtek RT5650 is what CELES & BANON have). I looked through some of the other included configs and saw there was a line like this at the top of the file: `Syntax 2` - adding this to the rt5650 config and running UCM fixed it.

The chromebook_audio packages just add UCM configs to the system; if the confits are broken, there is no use in having them.

**Also, read the docs. If I had a chromebook with the board name rammus and I wanted to run brunch on it, I would use that option. You don't use that option if your chromebook is different than the actual chromebook the recovery image is being used for**

Edit - native-chromebook-image straight out disables all the UCM2 configs, falling back on the UCM1 (chromebook_audio) configs. I submitted a PR to fix UCM2 and it will be included in brunch 102 so that option is no longer needed to use the audio.
(Native-chromebook-image broke my touchpad)
